docs/operators/kinesisInputOperator.md

```diff
@@ -0,0 +1,178 @@
+
+#### Ports
+
+`outputPort <T>`: Tuples extracted from Kinesis streams in form of records are emitted through this port.
+
+### KinesisConsumer
+
+This is an abstract implementation of Kinesis consumer. It sends the fetch
+requests from AWS kinesis streams to Kinesis partitions. For each request,
+it receives the set of records and stores them into the buffer which is
+ArrayBlockingQueue.
+
+
+### ShardManager
+
+This is an interface for offset management and is useful when consuming data
+from specified offsets. Updates the offsets for all the kinesis partitions
+periodically. Below is the code snippet:
+
+```java
+public class ShardManager
+{
+  public Map<String, String> loadInitialShardPositions();
+  public Map<String, String> loadInitialShardPositions();
+  public void updatePositions(Map<String, String> shardPositions);
+}
+```
+
+### Partitioning
+
+The logical instance of the KinesisInputOperator acts as the Partitioner
+as well as a StatsListener. This is because the
+AbstractKinesisInputOperator implements both the
+com.datatorrent.api.Partitioner and com.datatorrent.api.StatsListener
+interfaces and provides an implementation of definePartitions(...) and
+processStats(...) which makes it auto-scalable.
+
+#### Response processStats(BatchedOperatorStats stats)
+
+The application master invokes this method on the logical instance with
+the stats (tuplesProcessedPS, bytesPS, etc.) of each partition.
+Re-partitioning happens based on whether any new Kinesis streams are added/modified 
+or based on kinesis shards stats with respective upper bounds.
+
+#### DefinePartitions
+
+Based on the repartitionRequired field of the Response object which is
+returned by processStats(...) method, the application master invokes
+definePartitions(...) on the logical instance which is also the
+partitioner instance. Dynamic partition can be disabled by setting the
+parameter repartitionInterval value to a negative value.
+
+
+#### KinesisStringInputOperator
+This class extends from AbstractKinesisInputOperator and defines the `getTuple()` method which extracts byte array from Kinesis record.
+
+#### Ports
+`outputPort <byte[]>`: Tuples extracted from Kinesis Streams are emitted through this port.
+
+### Application Example
+
+This section builds an Apex application using Kinesis input operator.
+Below is the code snippet:
+
+```java
+@ApplicationAnnotation(name="KinesisApp")
+public class ExampleApplication implements StreamingApplication
+{
+
+  @Override
+  public void populateDAG(DAG dag, Configuration conf)
+  {
+    KinesisStringInputOperator inputOperator = dag.addOperator("kinesisInput", new KinesisStringInputOperator());
+    ConsoleOutputOperator outputOperator = dag.addOperator("Console", new ConsoleOutputOperator());
+    dag.addStream("kinesis-to-console", inputOperator.outputPort, outputOperator.input);
+  }
+}
+```
+Below is the configuration for “KINESIS-STREAM-NAME” Kinesis stream name:
```

Comment:
Did not quite understand the meaning of this line. Are you referring to configuration for the app ?